crates/radix-engine-toolkit/tests/transaction_types.rs

```diff
@@ -2645,8 +2522,57 @@ fn lock_fee_manifest_has_no_classification_except_general() {
         dynamic_analysis.detailed_classification.len(),
         1
     );
-    assert_eq!(
-        static_analysis.classification,
-        indexset![ManifestClass::General]
-    );
+    assert_eq!(static_analysis.classification, indexset![
+        ManifestClass::General
+    ]);
+}
+
+#[test]
+fn subintent_manifest_of_transfer_is_a_general_subintent() {
+    let account =
+        ComponentAddress::new_or_panic([EntityType::GlobalAccount as u8; 30]);
+    subintent_manifest_classification_test(
+        ManifestBuilder::new_subintent_v2()
+            .lock_fee(account, dec!(10))
+            .withdraw_from_account(account, XRD, 10)
+            .take_all_from_worktop(XRD, "bucket")
+            .try_deposit_or_abort(account, None, "bucket")
+            .yield_to_parent(())
+            .build(),
+        true,
+    )
+}
+
+#[test]
+fn subintent_manifest_of_transfer_with_metadata_update_is_not_a_general_subintent()
+ {
+    let account =
+        ComponentAddress::new_or_panic([EntityType::GlobalAccount as u8; 30]);
+    subintent_manifest_classification_test(
+        ManifestBuilder::new_subintent_v2()
+            .lock_fee(account, dec!(10))
+            .withdraw_from_account(account, XRD, 10)
+            .take_all_from_worktop(XRD, "bucket")
+            .try_deposit_or_abort(account, None, "bucket")
+            .set_metadata(account, "some_key", "some_value")
+            .yield_to_parent(())
+            .build(),
+        false,
+    )
+}
+
+fn subintent_manifest_classification_test(
```

Comment:
SUPER MINOR/OPTIONAL: This logic is quite simple and I think this helper obscures the assertion...

In general, tests are special, in that they should be optimised for clarity and the correct level of specificity, and I'd suggest all test helpers should aid readability of tests (either set-up or assertions).

e.g. If we had a helper, I'd prefer it to be `fn manifest_has_class(&impl ReadableManifest, class: ManifestClass) -> bool` / `fn manifest_does_not_have_class(&impl ReadableManifest, class: ManifestClass) -> bool`.

So simple I think it's clearer to inline in the test to make the assertion more obvious.

Reply:
But honestly, checking for an exact list of classifications is even better I think - as the whole list is more useful for detecting regressions :)

e.g. `fn manifest_has_classes(manifest: &impl ReadableManifest, classes: impl IntoIterator<Item = ManifestClass>)`

Which can then be used like e.g.:

```
assert!(manifest_has_classes(manifest, [ManifestClass::GeneralSubintent]);
```